src/api/draftSpecs.ts

```diff
@@ -1,34 +1,67 @@
-import { insertSupabase, TABLES, updateSupabase } from 'services/supabase';
+import {
+    deleteSupabase,
+    insertSupabase,
+    TABLES,
+    updateSupabase,
+} from 'services/supabase';
 import { ENTITY } from 'types';
 
+interface CreateMatchData {
+    draft_id: string | null;
+    catalog_name: string;
+    spec: any;
+    spec_type?: ENTITY | null;
+    expect_pub_id?: string;
+}
+
+interface UpdateMatchData {
+    draft_id: string | null;
+    catalog_name: string;
+    expect_pub_id?: string;
+}
+
 export const createDraftSpec = (
     draftId: string | null,
     catalogName: string,
     draftSpec: any,
-    specType?: ENTITY | null
+    specType?: ENTITY | null,
+    lastPubId?: string | null
 ) => {
-    return insertSupabase(TABLES.DRAFT_SPECS, {
+    let matchData: CreateMatchData = {
         draft_id: draftId,
         catalog_name: catalogName,
         spec_type: specType ?? undefined,
         spec: draftSpec,
-    });
+    };
+
+    if (lastPubId) {
+        matchData = { ...matchData, expect_pub_id: lastPubId };
```

Comment:
The `expect_pub_id` column of the `draft_specs` table does not accept undefined or null values.